odoo/osv/expression.py

```diff
@@ -395,6 +395,55 @@ def check_leaf(element, internal=False):
     if not is_operator(element) and not is_leaf(element, internal):
         raise ValueError("Invalid leaf %s" % str(element))
 
+# --------------------------------------------------
+# Python utils
+# --------------------------------------------------
+
+# Python functions capable of evaluating a domain leaf.
+#
+# .. note:: The 'like' operator takes a compiled regexp instead and case
+#           sensitivity is handled by the regexp itself.
+#
+# :param a: left operand data in a list as returned by `mapped`
+# :param b: right operand value
+# :return: bool
+PY_OPERATORS = {
+    "=": lambda a, b: b in a,
+    "=?": lambda a, b: b is False or b in a,
+    "<": lambda a, b: any(x is not None and x < b for x in a),
+    "<=": lambda a, b: any(x is not None and x <= b for x in a),
+    ">": lambda a, b: any(x is not None and x > b for x in a),
+    ">=": lambda a, b: any(x is not None and x >= b for x in a),
+    "in": lambda a, b: b and any(x in b for x in a),
+    "like": lambda a, b: any(x and b.search(x) for x in a),
+    "ilike": lambda a, b: any(x and b.search(x) for x in a),
+    "=like": lambda a, b: any(x and b.fullmatch(x) for x in a),
+    "=ilike": lambda a, b: any(x and b.fullmatch(x) for x in a),
+}
+
+# Fill-in the missing negation operators
+for op in list(PY_OPERATORS):
+    negation = TERM_OPERATORS_NEGATION.get(op)
+    if negation and negation not in PY_OPERATORS:
+        PY_OPERATORS[negation] = lambda a, b, fn=PY_OPERATORS[op]: not fn(a, b)
+
+
+def compile_search_regex(search, ignorecase=False):
+    """Compile a regular expression for a search string.
```

Comment:
This should compile a LIKE pattern to a regular expression, nothing more.
```suggestion
def make_like_regex(pattern, ignorecase=False):
    """Compile a pattern for a LIKE comparison to a regular expression object.
```